test/testdata.go

```diff
@@ -1,17 +1,21 @@
 package test
 
 import (
+	"time"
+
+	"github.com/argoproj/gitops-engine/pkg/utils/testing"
 	apiv1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime"
 
-	"github.com/argoproj/gitops-engine/pkg/utils/testing"
-
 	"github.com/argoproj/argo-cd/common"
 	apps "github.com/argoproj/argo-cd/pkg/client/clientset/versioned/fake"
 	appinformer "github.com/argoproj/argo-cd/pkg/client/informers/externalversions"
 	applister "github.com/argoproj/argo-cd/pkg/client/listers/application/v1alpha1"
+	servercache "github.com/argoproj/argo-cd/server/cache"
+	cacheutil "github.com/argoproj/argo-cd/util/cache"
+	appstatecache "github.com/argoproj/argo-cd/util/cache/appstate"
```

Comment:
I think these are responsible for the import cycle. Imho, the `NewServerInMemoryCache()` should be moved to `cluster_test.go`, as a non-exported function (or, in the long run, split up package `test` into more specific test fixture packages).

Reply:
thank you! fixed.